rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -25,10 +26,28 @@
 
 from sawtooth_rest_api.protobuf.validator_pb2 import Message
 
-
 LOGGER = logging.getLogger(__name__)
 
 
+class _Backoff:
+    """Implements a simple exponential backoff mechanism.
+    """
+
+    def __init__(self):
+        self.num_retries_left = 3
+        self.min_interval = 0.1
+        self.max_interval = 1.0
+        self.interval = self.min_interval
+
+    async def do_backoff(self):
+        if not self.num_retries_left:
+            LOGGER.warning("No more retries left. Backoff terminated...")
+            raise DisconnectError()
+
+        self.interval = random.randrange(self.interval, self.max_interval)
```

Comment:
There is probably no need for a random interval here, as the goal of this back off is to avoid bombarding the validator with requests, not to avoid submitting at the same time as another component.

Reply:
The interval increases with each retry. The interval value is selected from range that goes from the previous interval value to the max. We keep backing off for slightly longer periods with each retry.

Reply:
Why not just use a predictable increase, like `* 2`?